redis/asyncio/client.py

```diff
@@ -759,7 +764,12 @@ async def _disconnect_raise_connect(self, conn, error):
         if retry_on_timeout is not set or the error
         is not a TimeoutError. Otherwise, try to reconnect
         """
-        await conn.disconnect()
+        try:
+            await asyncio.shield(conn.disconnect())
```

Comment:
I think this leads to race conditions since it doesn't guarantee that it will be disconnected.